aws-xray-agent-plugin/build.gradle.kts

```diff
@@ -77,6 +77,42 @@ tasks {
         rename("disco-java-agent-$discoVer.jar", "disco-java-agent.jar")
     }
 
+    // JARs are just archives, so we can unzip it to a tmp folder
```

Comment:
This stackoverflow seems (?) like it could work without unzipping the whole jar and less code

https://stackoverflow.com/questions/60433224/read-manifest-from-jar-with-gradle

Reply:
It looks like it would work for reading, but not for writing. `fromArchiveEntry` returns a read-only object: https://gradle.github.io/kotlin-dsl-docs/api/org.gradle.api.resources/-text-resource/index.html

Reply:
Yeah we'd still want to read from the `zipTree`, filtering out the MANIFEST, in the final archiving task. But it'd avoid having the temp directory I think.